node/src/accountant/db_access_objects/failed_payable_dao.rs

```diff
@@ -0,0 +1,778 @@
+                }
+                Err(e) => {
+                    return Err(FailedPayableDaoError::SqlExecutionFailed(e.to_string()));
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    fn delete_records(&self, hashes: &HashSet<TxHash>) -> Result<(), FailedPayableDaoError> {
+        if hashes.is_empty() {
+            return Err(FailedPayableDaoError::EmptyInput);
+        }
+
+        let hashes_vec: Vec<TxHash> = hashes.iter().cloned().collect();
+        let sql = format!(
+            "DELETE FROM failed_payable WHERE tx_hash IN ({})",
+            comma_joined_stringifiable(&hashes_vec, |hash| { format!("'{:?}'", hash) })
+        );
+
+        match self.conn.prepare(&sql).expect("Internal error").execute([]) {
+            Ok(deleted_rows) => {
+                if deleted_rows == hashes.len() {
+                    Ok(())
+                } else if deleted_rows == 0 {
+                    Err(FailedPayableDaoError::NoChange)
+                } else {
+                    Err(FailedPayableDaoError::PartialExecution(format!(
+                        "Only {} of the {} hashes has been deleted.",
+                        deleted_rows,
+                        hashes.len(),
+                    )))
+                }
+            }
+            Err(e) => Err(FailedPayableDaoError::SqlExecutionFailed(e.to_string())),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::accountant::db_access_objects::failed_payable_dao::FailureReason::{
+        NonceIssue, PendingTooLong,
+    };
+    use crate::accountant::db_access_objects::failed_payable_dao::FailureRetrieveCondition::ByReason;
+    use crate::accountant::db_access_objects::failed_payable_dao::{
+        FailedPayableDao, FailedPayableDaoError, FailedPayableDaoReal, FailureReason,
+        FailureRetrieveCondition,
+    };
+    use crate::accountant::db_access_objects::test_utils::FailedTxBuilder;
+    use crate::blockchain::test_utils::make_tx_hash;
+    use crate::database::db_initializer::{
+        DbInitializationConfig, DbInitializer, DbInitializerReal, DATABASE_FILE,
+    };
+    use crate::database::rusqlite_wrappers::ConnectionWrapperReal;
+    use crate::database::test_utils::ConnectionWrapperMock;
+    use masq_lib::test_utils::utils::ensure_node_home_directory_exists;
+    use rusqlite::{Connection, OpenFlags};
+    use std::collections::{HashMap, HashSet};
+    use std::str::FromStr;
+
+    #[test]
+    fn insert_new_records_works() {
+        let home_dir =
+            ensure_node_home_directory_exists("failed_payable_dao", "insert_new_records_works");
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let tx1 = FailedTxBuilder::default().hash(make_tx_hash(1)).build();
+        let tx2 = FailedTxBuilder::default()
+            .hash(make_tx_hash(2))
+            .reason(PendingTooLong)
+            .checked(1)
```

Comment:
I'm not a fan of this either. 

You're creating unrealistic conditions for this test. 
In the real world we'll never insert a record with `checked` already in. 
It's a question for you if we want to test it before going off to the attempt.